### PR TITLE
#208: Refactored endpoint functions into structs

### PR DESCRIPTION
**Pull request by bonkey**
For review just look at 8c8efd5673af04bf1ca10a6f03c613a43025cdff.

Rest will be merged by @ahmed-shehata "in 10-15 minutes" counting from 30 minutes ago.

### CODE REVIEW

**Comment by ahmed-shehata** [comment]
.. After receiving @bonkey's mighty 👍 on https://github.com/zalando-incubator/atlas-ios/pull/209 😉

File: AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+extension Endpoint {
+
+    var method: HTTPMethod { return .GET }
+
+    var contentType: String { return "application/json" }
+    var acceptedContentType: String { return "application/json" }
+
+    var queryItems: [NSURLQueryItem]? { return nil }
+    var parameters: Dictionary<String, AnyObject>? { return nil }
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift]: why not `[String: AnyObject]?`?

File: AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+}
+
+extension Endpoint {
+
+    var description: String {
+        let params: String = {
+            guard let parameters = parameters else {
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift]: one line?

File: AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift (hunk 1 above)
Comment (bonkey) [inline AtlasSDK/AtlasSDK/Endpoints/Endpoint.swift]: Well, it was like this. This is a leftover after some experiments. Will fix it in another PR

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
Or actually, before 👍 I'll add more tests

**Comment by bonkey** [comment]
👍

File: AtlasSDK/AtlasSDK/PublicAPIClient.swift [hunk 3]
```diff
@@ -32,17 +32,24 @@ public typealias CheckoutCompletion = AtlasResult<Checkout> -> Void
  */
 public typealias OrderCompletion = AtlasResult<Order> -> Void
 
+@available( *, deprecated, message = "Should be deleted")
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/PublicAPIClient.swift]: what is deprecated?

Comment (bonkey) [inline AtlasSDK/AtlasSDK/PublicAPIClient.swift]: Because it should rather be handled differently, not with some fake Empty result

Comment (bonkey) [inline AtlasSDK/AtlasSDK/PublicAPIClient.swift]: true, that it should be `EmptyResponse` actually, but was easier to mark this one :)

**Comment by hani-ibrahim** [comment]
👍

**Comment by bonkey** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**